keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ArtifactSupplier.kt

```diff
@@ -0,0 +1,27 @@
+package com.netflix.spinnaker.keel.api.plugins
+
+import com.netflix.spinnaker.keel.api.events.ArtifactEvent
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.kork.plugins.api.internal.SpinnakerExtensionPoint
+
+/**
+ * Keel plugin interface to be implemented by suppliers of published artifact information.
+ *
+ * The job of an [ArtifactSupplier] is to detect new versions of artifacts, using
+ * whatever mechanism they choose (e.g. they could receive events from another system,
+ * or poll an artifact repository for artifact versions), and notify keel via the [publishArtifact]
+ * method, so that the artifact versions can be persisted and evaluated for promotion.
+ *
+ * The default implementation of [publishArtifact] simply publishes the event via the [EventPublisher],
+ * and should *not* be overridden by implementors.
+ */
+interface ArtifactSupplier : SpinnakerExtensionPoint {
```

Comment:
The new plugin interface, hopefully self-explanatory from the code and documentation above.

Reply:
nit: Would rename to `ArtifactPublisher`, since this interface isn't supplying anything (as "supplier" traditionally has implied in Java's std lib taxonomy).

Reply:
Good point. Done!